### PR TITLE
Cleanup dependencies

### PR DESCRIPTION
**Pull request by tokengeek**
Tweaking the Gemfiles and travis settings in order to make
the dependencies as open as possible.

Limiting `nokogiri` for older rubies caused problems for users with
Rails 4.0 applications so we should prioritise supported versions
of Ruby and document workarounds.

### CODE REVIEW

**Comment by tokengeek** [comment]
Hoping to add some documentation before merging, then removing
the dependency added to the `gemspec`

Unfortunately I can see we either update all the Gemfiles and travis.yml
which is the right thing in my opinion, or add a restriction for newer Rubies
which is a simpler (short term fix).

File: .travis.yml [hunk 1]
```diff
@@ -19,26 +11,21 @@ script: bundle exec rake travis
...
-      gemfile: Gemfile
-    - rvm: 1.9.2
-      gemfile: Gemfile.1.8.7
-    - rvm: 1.9.3
       gemfile: Gemfile.1.8.7
-    - rvm: 2.0.0
+    - rvm: jruby-18mode
       gemfile: Gemfile.1.8.7
+    - rvm: 1.9.2
+      gemfile: Gemfile.1.9.3
```
Comment (plribeiro3000) [inline .travis.yml]: I believe we just dropped support to `1.9.2` at fog `2.0.0`. Could we remove it here?

Comment (tokengeek) [inline .travis.yml]: Yep. We probably should. I'll update.

**Comment by icco** [comment]
@tokengeek + @plribeiro3000 I merged master into this and updated travis to be closer to Fog. Thoughts?

**Comment by plribeiro3000** [comment]
LGTM!

**Comment by plribeiro3000** [comment]
Travis seems to be having problems building the ruby binaries. =s

**Comment by geemus** [comment]
Bringing this in. Looks like there is a broken test, but that it was pre-existing condition.

**Pull request merged**